### ISSUE TITLE
conflit with More Character Portrait (Vanilla)

### ISSUE THREAD

**Issue by bishem** [SirHartley/Industrial.Evolution#1]
i tried finding some guidelines of how you want your issues written, but found none, so sorry in advance if it's a little messy

the variant vanilla of the mod [more character portraits](https://www.nexusmods.com/starsector/mods/100) is conflicting with yours

what i gathered is that images from the game's data/graphics/portraits folder are (at least) used in files inside your mod's data/world/faction that are overridden by the mod above

i don't know if it is fixable but any portrait or image mod that overrides those you use in your mod will likely conflict with it

i am but a humble developer that knows almost nothing of the modding process of this game so the only workaround i found was to backup the game's current portrait folder, to then copy the entire content of more character portraits over it, which is admittedly quite dirty

if you have an idea of how to resolve it, huzzah, same if it helped in any other way

if not this is a moot point and can be safely ignored

in any case i wish you good luck on your future endeavours

**Comment by SirHartley** [comment]
Good evening, 

Thank you for the report! I'm not entirely certain what this conflict entails - do the IndEvo files take precedence and cause the MCP portraits to not be displayed? Or do the MCP portraits override the ones in my own mod? 

I'm slightly confused since the factions in this mod are mostly of a technical nature and shouldn't be visible within the game.

If you have a minute to let me know of the exact problem you're facing, I'd appreciate it.
Thanks!
-H

**Comment by bishem** [comment]
the files from IndEvo do not override any files from MCP, i did not see anything overlapping between the two which is the first thing i noticed

i then found in [deconStorageColour.faction](https://github.com/SirHartley/Industrial.Evolution/blob/master/data/world/factions/submarkets/deconStorageColour.faction) a line referencing "graphics/portraits/portrait_hegemony02.png"
which is a file existing in the game's core data that MCP do overrides

my understanding (and bear in mind this is highly speculative on my end) is that when the game launches it somehow replaces the references it has of its core files by the ones it can find in installed mods

in this case "graphics/portraits/portrait_hegemony02.png" does exists in both the game core data and MCP but for some reason only the files on MCP are now seen by the game

the result being that upon launching the game a alert pops up saying IndEvo is pointing to null, i would have attached the log file but it contains no ERROR line with is very peculiar to me but, it is what it is i guess i will try to get more data to send your way

give me 2 min

**Comment by bishem** [comment]
unfortunately i do not seem to be able to reliably reproduce the bug

i tried with IndEvo and MCP alone and with my entire modlist, it does not crash anymore

it did crash once or twice when i started testing with the alert below

![image](https://github.com/user-attachments/assets/85da2efa-dbc4-4b84-8db6-f41dbd50aeb3)

i also found this in my logs (i don't know what i was looking at before -_-)
it does not seem to print out the full stacktrace though
```log
28002 [Thread-3] ERROR lunalib.lunaSettings.LunaSettings  - LunaSettings: Could not find mod IndEvo
28063 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.ExceptionInInitializerError
java.lang.ExceptionInInitializerError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at com.fs.starfarer.campaign.rules.oOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO.getCommandClass(Unknown Source)
	at com.fs.starfarer.campaign.rules.oOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO.<init>(Unknown Source)
	at com.fs.starfarer.campaign.rules.Rules.o00000(Unknown Source)
	at com.fs.starfarer.loading.SpecStore.�O0000(Unknown Source)
	at com.fs.starfarer.loading.ResourceLoaderState.init(Unknown Source)
	at com.fs.state.AppDriver.begin(Unknown Source)
	at com.fs.starfarer.combat.CombatMain.main(Unknown Source)
	at com.fs.starfarer.StarfarerLauncher.super(Unknown Source)
	at com.fs.starfarer.StarfarerLauncher$1.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at indevo.utils.helper.Settings.getInt(Settings.java:126)
	at indevo.industries.salvageyards.rules.IndEvo_InitSYCustomProductionDiag.<clinit>(IndEvo_InitSYCustomProductionDiag.java:71)
	... 15 more
```

i guess i was just unlucky because every time i tried both together it failed and after doing what i said in OP it worked T^T

**Comment by bishem** [comment]
i just realized that i forgot to send you my mod list

this is the bare minimum i used for testing
```json
{"enabledMods": [
  "lw_console",
  "aaacrew_replacer",
  "IndEvo",
  "lw_lazylib",
  "lunalib",
  "MagicLib",
  "more_character_portraits",
  "more_character_portraits_vanilla",
  "particleengine",
  "RetroLib",
  "variants_lib",
  "shaderLib"
]}
```

this is the full list
```json
{"enabledMods": [
  "pantera_ANewLevel40R",
  "advanced_gunnery_control_dbeaa06e",
  "Planetace_AstralAscension",
  "lw_autosave",
  "timid_admins",
  "CaptainsLog",
  "chatter",
  "lw_radar",
  "lw_console",
  "timid_cum",
  "aaacrew_replacer",
  "diyplanets",
  "dynamictariffs",
  "fleethistory",
  "fleet_journal",
  "fleetsizebydp",
  "sun_fuel_siphoning",
  "GrandColonies",
  "guarantee-rare-items",
  "hostileIntercept",
  "illustrated_entities",
  "immersionFriendlyPortraitPack",
  "IndEvo",
  "largerZoomOut",
  "lw_lazylib",
  "leadingPip",
  "logisticsNotifications",
  "lunalib",
  "MagicLib",
  "More_Ai_Core_Skills",
  "more_character_portraits",
  "more_character_portraits_vanilla",
  "more_procgen_names",
  "more_ship_names",
  "sun_new_beginnings",
  "nexerelin",
  "officerExtension",
  "particleengine",
  "planet_search",
  "wyv_planetaryShieldAccessControl",
  "player_portraits_extended",
  "portrait_changer",
  "progressiveSMods",
  "pt_qolpack",
  "QualityCaptains",
  "RetroLib",
  "scan_those_gates",
  "speedUp",
  "stelnet",
  "transfer_all_items",
  "variants_lib",
  "WEAPONARCS",
  "whichmod",
  "whichtmi",
  "shaderLib"
]}
```

**Comment by SirHartley** [comment]
This crash is not caused by an incompatibility with a portrait pack. It's caused by:

```
28002 [Thread-3] ERROR lunalib.lunaSettings.LunaSettings  - LunaSettings: Could not find mod IndEvo
Caused by: java.lang.NullPointerException at indevo.utils.helper.Settings.getInt(Settings.java:126)
```

The mod attempting to fetch a lunaSettings int, failing to find lunalib, falling back on settings.json, failing to find the value because I did not add it since lunalib is a required dependency and I stopped updating the vanilla settings file, and crashing.

Kindly re-download industrial evolution, it is likely you have a corrupted version.
Let me know if the issue persists.
